src/extension.ts

```diff
@@ -197,6 +198,25 @@ export async function activate(context: vscode.ExtensionContext) {
     addDecorations();
   };
 
+  function logBranchTypes(event: vscode.TextEditorSelectionChangeEvent) {
+    const getBranch = (branch: SyntaxNode[]): SyntaxNode[] => {
+      if (branch[0].parent) {
+        return getBranch([branch[0].parent, ...branch]);
+      }
+      return branch;
+    };
```

Comment:
whoah recursion.  this one is a bit tough to follow 😅.  would `getAncestors` be a better name?  Presuming I've understood the function correctly...

I think this one might be a bit clearer without recursion too.  Just iterate until the parent is null and add the nodes as you go up.  I also prefer `== null` comparisons to `boolean` transformations

Also I generally try to avoid nesting functions unless it needs to capture scope

Reply:
Yeah I hear you, this was mostly meant for me at the time of writing. This is probably an artifact of the fact that my first intro to programming around syntax trees was in SML & Racket 😆 I can clean this one up and move it to a separate file.